pkg/util/tradingutil/orders.go

```diff
@@ -0,0 +1,53 @@
+package tradingutil
+
+import (
+	"fmt"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func HasSufficientBalance(
+	baseCurrency string,
+	quoteCurrency string,
+	side types.SideType,
+	price, quantity fixedpoint.Value,
+	tradingMarket types.Market,
+	balances types.BalanceMap,
+) (bool, string) {
+	// 1. check if the currency is correct
+	_, okBase := balances[baseCurrency]
+	_, okQuote := balances[quoteCurrency]
+	if !okBase || !okQuote {
+		return false, fmt.Sprintf("invalid currency paire: %s/%s", baseCurrency, quoteCurrency)
```

Comment:
maybe return type error? and use fmt.Errorf() for the error message

Reply:
ok